nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/recurrent/LSTMLayer.java

```diff
@@ -76,33 +74,39 @@
  */
 public class LSTMLayer extends DynamicCustomOp {
 
-    private LSTMConfiguration configuration;
+    private LSTMLayerConfig configuration;
 
     @Getter
-    private LSTMWeights weights;
+    private LSTMLayerWeights weights;
+
 
     public LSTMLayer() {
     }
 
-    public LSTMLayer(@NonNull SameDiff sameDiff, SDVariable maxTSLength, SDVariable x, SDVariable cLast, SDVariable yLast, LSTMWeights weights, LSTMConfiguration configuration) {
-        super(null, sameDiff, weights.argsWithInputs(maxTSLength, x, cLast, yLast));
+    public LSTMLayer(@NonNull SameDiff sameDiff, SDVariable maxTSLength, SDVariable x, SDVariable cLast, SDVariable yLast, LSTMLayerWeights weights, LSTMLayerConfig configuration) {
+        super(null, sameDiff, weights.argsWithInputs(x, maxTSLength, cLast, yLast));
         this.configuration = configuration;
         this.weights = weights;
-        addIArgument(configuration.iArgs(true));
-        addTArgument(configuration.tArgs());
+        addIArgument(iArgs());
+        addTArgument(tArgs(configuration));
+        addBArgument(bArgs(weights, maxTSLength, yLast, cLast, configuration));
```

Comment:
no need to have configuration object in the signature for tArgs and bArgs methods, when we can access it from the configuration field (like iArgs method already does)